crates/worktree/src/worktree.rs

```diff
@@ -987,11 +1000,35 @@ impl LocalWorktree {
         !self.share_private_files && self.settings.is_path_private(path)
     }
 
-    fn restart_background_scanners(&mut self, cx: &mut ModelContext<Worktree>) {
+    fn restart_background_scanners(
+        &mut self,
+        cx: &mut ModelContext<Worktree>,
+        force_reindex: bool,
+    ) {
         let (scan_requests_tx, scan_requests_rx) = channel::unbounded();
         let (path_prefixes_to_scan_tx, path_prefixes_to_scan_rx) = channel::unbounded();
         self.scan_requests_tx = scan_requests_tx;
         self.path_prefixes_to_scan_tx = path_prefixes_to_scan_tx;
+        if force_reindex {
```

Comment:
Why does changing our always_include settings require us to re-create the entire snapshot? Shouldn't we just queue the matching paths for re-scanning via the `scan_requests_tx`?

Reply:
Previous paths that were always_included will get left in the index, so if someone changes their setting, those old entries will still be left around in addition to the new ones that got added. An example might be if someone adds a directory to their always_included setting and realizes it's clogging up their search results (looking at you node_modules). Without the reindex, all of those entries will still be around until the user quits Zed and forces it to reindex anyways.
One other approach that I thought about but never tried to implement was keeping around a list of which entries are always_included, then using that list to remove obsolete entries whenever the setting changes. The reason I went with this approach instead was:
1. It was a lot easier :)
2. It seemed like this setting wouldn't get changed very often and the reindexing was fast enough that it wasn't a problem.

I bet there are almost certainly better ways to go about doing this though. This is definitely an area of the code that I feel the hackiest about.

Reply:
Fair, I think keeping the old paths around, and then merging the new paths with the old paths and sending them all for re-indexing is a preferable solution here. It scales better for super-large projects (e.g. the Linux source), and allows us to leave supporting code as it is. + We  can avoid having a boolean argument :)

Reply:
@mikayla-maki Alright, I believe this should be fixed now!